src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -1019,26 +1019,27 @@ setBabelfishDependenciesForLogicalDatabaseDump(Archive *fout)
  * corresponding to specified logical database.
  */
 void
-addFromClauseForLogicalDatabaseDump(PQExpBuffer buf, TableInfo *tbinfo, bool is_builtin_db)
+addFromClauseForLogicalDatabaseDump(PQExpBuffer buf, TableInfo *tbinfo)
 {
 	if (strcmp(tbinfo->dobj.name, "babelfish_sysdatabases") == 0)
 	{
-		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d",
-						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
 		/*
-		 * builtin db will already be present in the target server so
-		 * no need to dump catalog entry for it.
+		 * Dump database catalog entry for specified logical database unless
+		 * it's a builtin database (dbid 1, 2, 3 or 4), in which case the db
+		 * will already be present in the target server so no need to dump
+		 * catalog entry for it.
 		 */
-		if (is_builtin_db)
-			appendPQExpBufferStr(buf, " LIMIT 0");
+		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d AND a.dbid > 4",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
```

Comment:
Shouldn't it be > 3 ?

Reply:
We have reserved 1 to 4 actually. refer: https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/contrib/babelfishpg_tsql/src/dbcmds.c#L308